pgspecial/dbcommands.py

```diff
@@ -564,6 +564,92 @@ def list_domains(cur, pattern, verbose):
         headers = [x[0] for x in cur.description]
         return [(None, cur, headers, cur.statusmessage)]
 
+@special_command('\\dF', '\\dF[+] [pattern]', 'List text search configurations.')
+def list_text_search_configurations(cur, pattern, verbose):
+
+    def _find_text_search_configs(cur, pattern):
+        sql = '''
+            SELECT c.oid,
+                 c.cfgname /*,
+                 n.nspname,
+                 p.prsname,
+                 np.nspname AS pnspname*/
+            FROM pg_catalog.pg_ts_config c
+            LEFT JOIN pg_catalog.pg_namespace n ON n.oid = c.cfgnamespace,
+                                                 pg_catalog.pg_ts_parser p
+            LEFT JOIN pg_catalog.pg_namespace np ON np.oid = p.prsnamespace
+            WHERE p.oid = c.cfgparser
+        '''
+
+        if pattern:
+            sql = cur.mogrify(sql + ' WHERE e.extname ~ {}'.format(pattern))
+
+        sql += ' ORDER BY 1'
+
+        log.debug(sql)
+        cur.execute(sql)
+        return cur.fetchall()
+
+    def _fetch_oid_details(cur, oid):
+        sql = '''
+            SELECT
+              (SELECT t.alias
+               FROM pg_catalog.ts_token_type(c.cfgparser) AS t
+               WHERE t.tokid = m.maptokentype ) AS "Token",
+                   pg_catalog.btrim( ARRAY
+                                      (SELECT mm.mapdict::pg_catalog.regdictionary
+                                       FROM pg_catalog.pg_ts_config_map AS mm
+                                       WHERE mm.mapcfg = m.mapcfg
+                                         AND mm.maptokentype = m.maptokentype
+                                       ORDER BY mapcfg, maptokentype, mapseqno ) :: pg_catalog.text, '{}') AS "Dictionaries"
+            FROM pg_catalog.pg_ts_config AS c,
+                 pg_catalog.pg_ts_config_map AS m
+            WHERE c.oid = %s
+              AND m.mapcfg = c.oid
+            GROUP BY m.mapcfg,
+                     m.maptokentype,
+                     c.cfgparser
+            ORDER BY 1;
+        '''
+
+        sql = cur.mogrify(sql, [oid])
+        log.debug(sql)
+        cur.execute(sql)
+
+        headers = [x[0] for x in cur.description]
+        return cur, headers, cur.statusmessage
+
+    if cur.connection.server_version < 80300:
+        not_supported = "Server versions below 8.3 do not support full text search."
+        cur, headers = [], []
+        return [(None, cur, None, not_supported)]
+
+    if verbose:
+        configs = _find_text_search_configs(cur, pattern)
+
+        for oid, cfgname in configs:
+            title = '\nObjects in extension "%s"' % cfgname
+            cur, headers, status = _fetch_oid_details(cur, oid)
+            results = (title, cur, headers, status)
+            yield results
```

Comment:
This `yield` statement is causing a bug and it's killing me.

> local variable 'is_special' referenced before assignment

Clearly that logic falls under `if verbose`, which is True by passing `+` to `\dF` as in `\dF+`, yet it causes the error when using the command _without_ the verbose flag.

We need to yield the results or else the same bug currently going on with `\dx+` will happen, where it will describe only the first result and leave the rest empty.

I hope I'm explaining myself, I'm having a bit of a hard time putting it into words. Let me know otherwise.

Reply:
Basically if you remove the `yield`, only `\dF` works. If you leave it, only `\dF+` works.

Reply:
It's hard to figure out what's going on here from reading the code, but have you tried changing all `return` places inside this function to `yield`? Mixing `return` and `yield` in the same function never works right. Basically instead of `return [(a, b, c, d)]` you'd put `yield a, b, c, d` everywhere.

Reply:
Well you learn something every day I guess :) Thanks for the tip, it worked.
I'll have a final version ready today.